src/mpi/coll/ireduce_scatter/ireduce_scatter.c

```diff
@@ -367,7 +367,7 @@ int MPIR_Ireduce_scatter(const void *sendbuf, void *recvbuf, const int recvcount
     void *in_recvbuf = recvbuf;
     void *host_sendbuf;
     void *host_recvbuf;
-    int count = recvcounts[MPIR_Comm_rank(comm_ptr)];
+    int count = MPIR_Comm_size(comm_ptr) * recvcounts[MPIR_Comm_rank(comm_ptr)];
```

Comment:
@tarudoodi I think this is assuming uniform count for all elements in the recvcounts array. I think to be correct we need to sum all of the elements. What do you think?

Reply:
I pushed a fixup patch for this that can be squashed if we agree.

Reply:
Yes, that would be more precise, in the current fix it may allocate more memory for some cases. And I did not think of this earlier but in current case if count is zero, it may fail again.
In all cases though, we will be allocating more memory for host_recvbuf, is that ok?

Reply:
Yes, we might over allocate the recvbuf in this case. I think that's OK for now. We likely have to adjust the helper API to include both a sendcount and recvcount to avoid overallocation.

Reply:
Yeah, I also think its ok for now.